modules/unsupported/mongodb/README.rst

```diff
@@ -2,17 +2,128 @@ MongoDB DataStore
+           "city": "Portland",
+           "year": "2014"
+           "attendance": "840"
+       }
+   }
+
+Is described using the following schema::
+
+   {
+       "typeName": "places",
+       "userData": {
+           "collection": "places"
+        },
+       "geometryDescriptor": {
+           "localName": "location",
+           "crs": {
+               "properties": {
+                   "name": "urn:ogc:def:crs:EPSG:4326"
+               },
+               "type": "name"
+           }
+       },
+       "attributeDescriptors": [
+           {
+               "localName": "location",
+               "type": {
+                   "binding": "com.vividsolutions.jts.geom.Point"
+               },
+               "userData": {
+                   "encoding": "GeoJSON",
+                   "mapping": "geometry"
+               }
+           },
+           {   "localName": "city",
+               "type": { "binding": "java.lang.String" },
+               "userData": { "mapping": "properties.name" }
+           },
+           {   "localName": "year",
+               "type": { "binding": "java.lang.String" },
+               "userData": { "mapping": "properties.year" }
+           },
+           {   "localName": "attendance",
+               "type": { "binding": "java.lang.String" },
+               "userData": {  "mapping": "properties.attendance" }
+           }
+       ]
+   }
+   
+   
+File URI schema stores:
+
+* For the directory-based schema store, edit the JSON document with the typeName requiring modification.
+  
+  Schemas are written createSchema() without indenting, but you can indent the resulting file for readability if desired.
+
+MongoDB URI schema stores:
+
+* Using a MongoDB document manipulation tool, update or insert the schema document in the collection maintaining the document in a form that follows the JSON schema format.
+  
+The JSON files contained in the file schema store are in a format that can be inserted into a MongoDB schema store (as long as the typeName in the file is unique to the document collection ).
+
+Multiple schemas, or views, can be created for a single MongoDB document collection by creating a new, unique, typeName and specifying the collection under the root-level userData object.
+
+Implementation Notes
+--------------------
+
+* Bounding box calculation makes use of a full table scan.
+
+* Multigeometry support requires MongoDB versions 2.5 and newer
+
+* Self-intersecting polygons is a common data problem preventing MongoDBDataStore from functioning. Please note that self-intersections may arise due to the transformation to WGS84 coordinates (which is a necessary preliminary step for importing data into MongoDB), even if they didn't exist in the original dataset.
+
+* All 2dsphere indexes and spatial operations assume the WGS84 ellipsoid. All indexed GeoJSON data stored in a MongoDB document collection is assumed to be referenced with the WGS84 coordinate reference system.
```

Comment:
Subtle detail I noticed earlier in use of "ellipsoid" versus "datum".  I decided to skip commenting on this earlier but subsequent commits introduced the term "datum".  I would think "datum" is more correct as this includes the prime meridian definition and hence the 0 reference point for longitude.  Yes, a minor detail...